### PR TITLE
Stripe Mock integration

### PR DESCRIPTION
**Pull request by bowlofstew**
This pull request (PR) provides local testing support for running against [stripe-mock](https://github.com/stripe/stripe-mock).  As part of this work, it was debatable whether or not to utilize an environment set variable for the `STRIPE_API_KEY` or not.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/subhub/241)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,11 +149,82 @@ doit dynalite:stop
...
+]
+```
+
+#### Get Hub function version
+```
+curl --silent http://127.0.0.1:5001/v1/hub/version
+{
+  "BRANCH": "local",
+  "REVISION": "latest",
+  "VERSION": "latest"
```
Comment (sidler-mozilla) [inline README.md]: let's remove examples with BRANCH=local and REVISION or VERSION =latest

File: docker-compose.yml [hunk 2]
```diff
@@ -21,6 +21,7 @@ services:
         AWS_ACCESS_KEY_ID: "fake-id"
         AWS_SECRET_ACCESS_KEY: "fake-key"
         STRIPE_API_KEY: $STRIPE_API_KEY
+        SUPPORT_API_KEY: $SUPPORT_API_KEY
```
Comment (sidler-mozilla) [inline docker-compose.yml]: duplicate value; look 2 lines below

File: docker-compose.yml [hunk 3]
```diff
@@ -55,3 +56,7 @@ services:
     ports:
       - 4567
 
+  stripe:
+    image: stripemock/stripe-mock:latest
+    ports:
+      - "12112:12112"
```
Comment (sidler-mozilla) [inline docker-compose.yml]: maybe cfg value with default of these values?

File: src/sub/app.py [hunk 4]
```diff
@@ -22,6 +22,23 @@
...
+    try:
+        requests.get(f"http://dynalite:{CFG.DYNALITE_PORT}")
+        return True
+    except Exception as e:
+        pass
+    return False
+
+
+if is_container():
+    stripe.log = "debug"
```
Comment (sidler-mozilla) [inline src/sub/app.py]: does case matter? i would think DEBUG would make more sense; if it doesnt mater, consider changing it to uppercase

File: docker-compose.yml (hunk 3 above)
Comment (bowlofstew) [reply docker-compose.yml]: configuration for docker like an docker-compose `.env` file?

File: docker-compose.yml [hunk 5]
```diff
@@ -21,6 +21,7 @@ services:
         AWS_ACCESS_KEY_ID: "fake-id"
         AWS_SECRET_ACCESS_KEY: "fake-key"
         STRIPE_API_KEY: $STRIPE_API_KEY
```
Comment (marty331) [inline docker-compose.yml]: Why the duplicate SUPPORT_API_KEY?

Comment (bowlofstew) [reply docker-compose.yml]: Already resolved locally.

File: src/sub/app.py (hunk 4 above)
Comment (bowlofstew) [reply src/sub/app.py]: It is case insensitive.

File: docker-compose.yml [hunk 6]
```diff
@@ -55,3 +58,8 @@ services:
     ports:
       - 4567
 
+  stripe:
+    image: stripemock/stripe-mock:latest
+    container_name: stripe
+    ports:
+      - "$STRIPE_API_PORT:12112"
```
Comment (bowlofstew) [inline docker-compose.yml]: A note, `12112` is the default, exposed port by Stripe Mock.  To change this requires extra command line parameters and thusly this port was not extracted to configuraiton.

File: .gitignore [hunk 7]
```diff
@@ -132,6 +132,7 @@ node_modules/
 
 # src tarballs
 .src.tar.gz
+.subhub.tar.gz
```
Comment (sidler-mozilla) [inline .gitignore]: why?

File: dodo.py [hunk 8]
```diff
@@ -57,21 +57,26 @@
...
 def envs(sep=' ', **kwargs):
     envs = dict(
-        DEPLOYED_ENV=CFG.DEPLOYED_ENV,
-        PROJECT_NAME=CFG.PROJECT_NAME,
         BRANCH=CFG.BRANCH,
-        REVISION=CFG.REVISION,
-        VERSION=CFG.VERSION,
-        REMOTE_ORIGIN_URL=CFG.REMOTE_ORIGIN_URL,
+        DEPLOYED_BY=CFG.DEPLOYED_BY,
+        DEPLOY_DOMAIN=CFG.DEPLOY_DOMAIN,
```
Comment (sidler-mozilla) [inline dodo.py]: ugh dont break up the DEPLOYED_ group of three please

File: dodo.py [hunk 9]
```diff
@@ -328,6 +333,7 @@ def task_check():
     yield check_noroot()
     yield gen_prog_check('python3.7')
     yield gen_prog_check('yarn')
+    yield gen_prog_check('docker-compose')
```
Comment (sidler-mozilla) [inline dodo.py]: cool nice addition! ;)

File: dodo.py [hunk 10]
```diff
@@ -643,17 +649,18 @@ def task_tar():
...
-        '--exclude=.git',
+        '--exclude=sub/tests',
+        '--exclude=hub/tests',
+        '--exclude=shared/tests'
     ])
     for src in SRCS:
         ## it is important to note that this is required to keep the tarballs from
         ## genereating different checksums and therefore different layers in docker
-        cmd = f'cd {CFG.REPO_ROOT}/src/{src} && echo "$(git status -s)" > {CFG.REVISION} && tar cvh {excludes} . | gzip -n > {CFG.SRCTAR} && rm {CFG.REVISION}'
+        cmd = f'cd {CFG.REPO_ROOT} && cp -rf .git {CFG.REPO_ROOT}/src/{src} && cd {CFG.REPO_ROOT}/src/{src} && echo "$(git status -s)" > {CFG.REVISION} && tar cvh {excludes} . | gzip -n > {CFG.SRCTAR} && rm {CFG.REVISION}'
```
Comment (sidler-mozilla) [inline dodo.py]: what is this? why?

File: src/hub/Dockerfile [hunk 11]
```diff
@@ -16,16 +16,11 @@ ENV HUB_API_KEY=$HUB_API_KEY
...
-ENV PROJECT_NAME=subhub
-ENV REMOTE_ORIGIN_URL=git@github.com:mozilla/subhub.git
-
 EXPOSE $LOCAL_FLASK_PORT
 
-RUN mkdir -p /subhub/hub
-ADD .src.tar.gz /subhub/hub
-WORKDIR /subhub/
+RUN mkdir -p /subhub
+ADD .subhub.tar.gz /subhub/hub
```
Comment (sidler-mozilla) [inline src/hub/Dockerfile]: why subhub instead of sub and hub separately if you dont like src???

File: src/sub/app.py [hunk 12]
```diff
@@ -22,6 +22,24 @@
 
 logger = get_logger()
 
+
+def is_docker() -> bool:
```
Comment (bowlofstew) [inline src/sub/app.py]: Mirrored from https://github.com/sindresorhus/is-docker/blob/master/index.js

File: dodo.py (hunk 8 above)
Comment (bowlofstew) [reply dodo.py]: Sorry was just putting them in alpha order to be able to find the ones of interest.

File: dodo.py (hunk 10 above)
Comment (bowlofstew) [reply dodo.py]: Don't you want the version/branch/etc from `.git`

**Pull request merged**